### ISSUE TITLE
field `envoy.config.listener.v3.FilterChain.name` unexpected marked as WIP

### ISSUE THREAD

**Issue by howardjohn** [envoyproxy/envoy#21800]
**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via emailing
envoy-security@googlegroups.com where the issue will be triaged appropriately.**

*Title*: field `envoy.config.listener.v3.FilterChain.name` unexpected marked as WIP

*Description*:
```
 field 'envoy.config.listener.v3.FilterChain.name' is marked as work-in-progress. API features marked as work-in-progress are not considered stable, are not covered by the threat model, are not supported by the security team, and are subject to breaking changes. Do not use this feature without understanding each of the previous points.
```

is now logged. I believe this is incorrect.

The `name` field has been around for a long time and used in logging. The WIP nature of the field only applies when used in conjunction with the new `filter_chain_matcher`. It seems like we should just log on `filter_chain_matcher` usage, not `name` which already has valid non-WIP usages.

*Repro steps*:
Use the field

cc @kyessenov

**Comment by ggreenway** [comment]
It looks like it was changed from `#not-implemented-hide` to `work_in_progrss in #20110. From this report, it sounds like it was incorrectly marked `#not-implemented-hide` before because it was being used in logging. @kyessenov should we remove the `work_in_progress` flag on this field?

**Comment by kyessenov** [comment]
Ack, I was going to do that, but feel free to go ahead.
